lib/database/Database.ts

```diff
@@ -1,59 +1,103 @@
+import { EventEmitter } from "events";
 import * as Sequelize from "sequelize";
 
-import ConfigInterface from "./../interfaces/ConfigInterface";
+import ConfigInterface from "../interfaces/ConfigInterface";
+import SequelizeDatabase from "./SequelizeDatabase";
 
-export default class Database {
+export default class Database extends EventEmitter {
   private config: ConfigInterface;
   private fromMemory: boolean;
   private memStorage: object;
+  private database?: SequelizeDatabase;
+  private model?: Sequelize.Model<any, any>;
 
   constructor(config: ConfigInterface) {
+    super();
+
     this.config = config;
     this.memStorage = {};
 
-    if (this.config.database.fromMemory) {
+    if (this.config.postgres.fromMemory) {
       this.fromMemory = true;
     } else {
       this.fromMemory = false;
-      // TODO: create sequelize
+      this.setupDatabase();
     }
   }
 
   public async connect(): Promise<void> {
-    // TODO: connect to sequelize and sync models
+    if (this.database) {
+      try {
+        await this.database.setup();
+
+        this.model = await this.database.getModel("Content");
+      } catch (err) {
+        super.emit("error", `Error getting models: ${err.message}`);
+      }
+    }
   }
 
   public async set(key: string, content: string): Promise<void> {
-
     if (this.fromMemory) {
       this.memStorage[key] = content;
       return;
     }
 
-    // TODO: store with sequelize model
+    if (this.model) {
+      this.model.upsert({
```

Comment:
when we initially used upsert(), our postgres version (9.3) did not support UPSERT, thus sequelize emulated upsert via a custom SQL-function which turned out to be very costly. That turned out to be a problem when we tried to import (unthrottled) messages from kafka.
So if this upsert is being executed in a performance-critical path, we should double-check if recent sequelize/postgres fully support upsert.

Reply:
Not really critical.